internal/api/user.go

```diff
@@ -144,3 +144,38 @@ func (h *handler) DeleteUser(c *gin.Context) {
 
 	c.JSON(http.StatusOK, response.SuccessResponse(nil))
 }
+
+func (h *handler) ChangePassword(c *gin.Context) {
+	me, err := helper.Me(c)
+	if err != nil {
+		c.JSON(http.StatusNotFound, response.ErrorResponse(response.CodeDefault, response.UserNotFound))
+		return
+	}
+
+	var body struct {
+		OldPassword string `json:"password" binding:"required"`
+		NewPassword string `json:"newpassword" binding:"required" validate:"min=10"`
+	}
+
+	err = c.Bind(&body)
+	if err != nil {
+		c.JSON(http.StatusBadRequest, response.ErrorResponse(response.CodeDefault, response.FormError))
+		return
+	}
+
+	if !me.Authenticate(body.OldPassword) {
+		c.JSON(http.StatusBadRequest, response.ErrorResponse(response.CodeDefault, "Could not authenticate password"))
```

Comment:
Can you add a constant for the ErrorMessage in `internal/api/response/response.go`?